[ESM] Localize system messages

BrtqKr opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->
PR includes changes for updated handling and translating OldDot system messages.
### Fixed Issues
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/<issueID>`.
2. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/<issueID>#issuecomment-1369752925`

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<issueID>
$ https://github.com/Expensify/App/issues/<issueID(comment)>

Do NOT only link the issue number like this: $ #<issueID>
--->
$ https://github.com/Expensify/App/issues/42923
PROPOSAL:


### Tests
Some of the messages aren't available, but taking export to CSV action as an example:

1. Sign in to the NewDot using account A and create a workspace with account B
2. Submit expense 
3. Sign in to the OldDot using account A
4. Go to reports
5. Mark a checkbox next to the report consisting of the submitted expense
6. Export to CSV
7. Go back to the NewDot, sign in to the account B
8. After a moment a system message from the account A `exported this report to CSV` should be visible

1. Go to the settings => preferences => langugage
2. Change english to spanish
3. go back to the report
4. verify that the message is correctly localized


<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment. 

For example:
1. Click on the text input to bring it into focus
9. Upload an image via copy paste
10. Verify a modal appears displaying a preview of that image
--->

- [x] Verify that no errors appear in the JS console

### Offline tests
Since this is just another variant of the action, there's nothing special in terms of offline testing - the message should be available when the device reconnects. 
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
5. Upload an image via copy paste
6. Verify a modal appears displaying a preview of that image
--->

- [x] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x ] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android: mWeb Chrome</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: Native</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: mWeb Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>
<img src="https://github.com/Expensify/App/assets/45288762/1fd9770d-731d-4e82-9f3e-c32b6a03ab09" width="400" /> 
<img src="https://github.com/Expensify/App/assets/45288762/b310306a-ecb6-4c9e-ab24-14e793ba678d" width="400" />

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Desktop</summary>

<!-- add screenshots or videos here -->

</details>

BrtqKr committed:
translations wip

BrtqKr committed:
adjust params, cleanup

BrtqKr committed:
add spanish translations, update delayed message

BrtqKr committed:
update messages wip

BrtqKr committed:
apply types

BrtqKr committed:
adjust types

BrtqKr committed:
revert deprecated messages

BrtqKr committed:
cleanup

deetergp reviewed (approved):
Overall looks pretty good to me. Thanks! I had a couple comments inline. Let's address those, get testing steps added, and complete the checklist.

deetergp reviewed (changes requested):
I Meant to hit "Request changes" not "Approve."

BrtqKr committed:
cleanup

BrtqKr committed:
Merge remote-tracking branch 'origin/main' into brtqkr/42923-localize-system-messages

BrtqKr committed:
cleanup

BrtqKr committed:
cleanup

BrtqKr committed:
update tests and remove depreacted messages

BrtqKr committed:
remove utility function for checking modified fields

BrtqKr committed:
fix unused types

BrtqKr committed:
run prettier

deetergp commented:
@BrtqKr You should be able to at least test the Export to CSV part of this by creating a collect workspace with two users, submitting an expense from one user to the other, approving the report in OldDot, and exporting it to CSV.

deetergp commented:
There's one thing you'll need to fix in order for it to work:
```diff
diff --git a/src/libs/ReportActionsUtils.ts b/src/libs/ReportActionsUtils.ts
index 204fec860a..a4de4c544d 100644
--- a/src/libs/ReportActionsUtils.ts
+++ b/src/libs/ReportActionsUtils.ts
@@ -192,6 +192,10 @@ function getWhisperedTo(reportAction: OnyxInputOrEntry<ReportAction>): number[]
     const originalMessage = getOriginalMessage(reportAction);
     const message = reportAction?.message;

+    if (originalMessage == '[]') {
+        return [];
+    }
+
     if (!(originalMessage && 'whisperedTo' in originalMessage) && !(message && 'whisperedTo' in message)) {
         return [];
     }
```

deetergp commented:
https://github.com/Expensify/App/assets/766197/f42eec13-ce64-42ff-9386-30e993764c67

trjExpensify commented:
@parasharrajat just a heads up this is an important one to jump on reviewing ASAP for #wave-collect. Thanks!

BrtqKr committed:
add original message condition

BrtqKr commented:
@deetergp Is this condition meant for other messages? I've included the change, but export to CSV seems to be working even without that. Anyway, it seems to be working as expected

<img width="406" alt="image" src="https://github.com/Expensify/App/assets/45288762/7a4afb40-385c-4ef7-81ee-78557894d5fb">

<img width="422" alt="image" src="https://github.com/Expensify/App/assets/45288762/f7e28bd2-a337-4566-931e-f91a951dd94a">

deetergp commented:
@BrtqKr It turns out the issue with Export CSV was because of [a regression](https://github.com/Expensify/App/pull/40168) that another contributor introduced. If you never updated to that version or have updated since (it has been fixed) then it should not be necessary.

deetergp commented on src/CONST.ts:
```diff
@@ -658,12 +658,11 @@ const CONST = {
                 CREATED: 'CREATED',
                 DELEGATE_SUBMIT: 'DELEGATESUBMIT', // OldDot Action
-                DELETED_ACCOUNT: 'DELETEDACCOUNT', // OldDot Action
+                DELETED_ACCOUNT: 'DELETEDACCOUNT', // Deprecated OldDot Action
```
Let's leave that one be for now

deetergp commented on src/libs/ReportActionsUtils.ts:
```diff
@@ -582,17 +587,6 @@ function isReportActionDeprecated(reportAction: OnyxEntry<ReportAction>, key: st
         return true;
     }
 
-    const deprecatedOldDotReportActions: ReportActionName[] = [
-        CONST.REPORT.ACTIONS.TYPE.DELETED_ACCOUNT,
-        CONST.REPORT.ACTIONS.TYPE.REIMBURSEMENT_REQUESTED,
-        CONST.REPORT.ACTIONS.TYPE.REIMBURSEMENT_SETUP_REQUESTED,
-        CONST.REPORT.ACTIONS.TYPE.DONATION,
-    ];
-    if (deprecatedOldDotReportActions.includes(reportAction.actionName)) {
-        Log.info('Front end filtered out reportAction for being an older, deprecated report action', false, reportAction);
-        return true;
-    }
```
That is a valid point. Nice catch @parasharrajat. 

@BrtqKr let's add these back. The back end changes I spoke of earlier will prevent new ones from being created, but it won't prevent old ones from being shown, which is what this code did.

BrtqKr committed:
remove condition

BrtqKr committed:
review fixes

BrtqKr committed:
Merge remote-tracking branch 'origin/main' into brtqkr/42923-localize-system-messages

BrtqKr committed:
modify conditions

BrtqKr committed:
Merge remote-tracking branch 'origin/main' into brtqkr/42923-localize-system-messages

BrtqKr committed:
modify conditions

BrtqKr committed:
fix types

BrtqKr commented on src/libs/ReportActionsUtils.ts:
```diff
@@ -1194,13 +1183,57 @@ function isOldDotReportAction(action: ReportAction): boolean {
-function getMessageOfOldDotReportAction(reportAction: OnyxEntry<ReportAction>): string {
-    if (!Array.isArray(reportAction?.message)) {
-        return getReportActionText(reportAction);
+function getMessageOfOldDotReportAction({originalMessage, actionName}: OldDotOriginalMessage): string {
```
Let's just use message the old way based on the predefined array, the rest of them are defined based on the key. I've already adjusted the types and the conditions

deetergp commented:
@parasharrajat There a lot of flows here but another one you can try is:
1. On a workspace with Workflows and Add Approvals enabled, submit an expense from a submitter to an approver to the workspace chat
2. Navigate to the Concierge chat (or any other chat that isn't workspace or the expense report)
3. In an incognito browser, log in as the approver and approve the expense and wait a moment
4. Go back to the submitter's browser window
    - [ ] Verify that the expense chat doesn't show up in the LHN as unread
 5. Navigate to the expense chat
     - [ ] Verify that there is a already message saying the approver approved the amount
  6. In Settings, change the language
     - [ ] Verify that the message text is translated to whichever language you switched to

deetergp commented:
I haven't tested this one yet, but it should work…
Given a workspace with an owner and an admin
1. Log into NewDot as the admin & verify you see an `#admins room`. If it is bolded/unread mark it unread, then navigate to a different chat.
3. Log into OldDot as the owner and navigate to the workspace settings.
4. Click Reports and add a named field and save it.
5. Then change the field name and save it.
6. Switch back to the admin in NewDot and wait a moment.
    - [ ] Verify that the #admins room does not go unread
7. Click into the `#admins` room
    - [ ] Verify that there is a message there showing that the field was changed
8. Change the admin's langage in Preferences
    - [ ] Verify that the message is translated

deetergp commented:
Those ought to be enough prove that this works.

BrtqKr commented:
@parasharrajat looks like it might be because of this thing https://github.com/Expensify/App/pull/43079#issuecomment-2188400627. [@deetergp  mentioned that it's already been fixed](https://github.com/Expensify/App/pull/43079#issuecomment-2189176386), so I assume this might still be relevant.

deetergp commented:
@parasharrajat If you haven't pulled `main` into your testing branch recently, give that as shot. The bug that introduce that _should_ be resolved now.

deetergp commented:
@BrtqKr I think you've got merge conflicts with the current `main`.

deetergp commented:
@parasharrajat My apologies, that is on me. I was working on that flow for a different PR. Neither `APPROVED` nor `SUBMITTED` are a part of this PR, so it is expected that neither of those would pass. Let me get you a different test case.

deetergp commented:
0. Have a Workflows + Approvals enabled workspace with a submitter and an approver
1. In OldDot, as the admin/approver go to Settings -> Workspaces -> your workspace -> Reports and add a field
2. In NewDot, as the submitter, submit an expense to the workspace chat
3. In OldDot, view the report and a value to the field
4. Pause a moment
5. In NewDot, view the expense report
    - [ ] Verify that a message is there saying the field was changed from something to something else
    - [ ] Verify that the message changes to Spanish when the language pref is changed

<img width="1728" alt="Screenshot 2024-06-27 at 12 14 00 PM" src="https://github.com/Expensify/App/assets/766197/2d7ace9c-b245-49f9-bcfc-261a9b1569c6">
<img width="855" alt="Screenshot 2024-06-27 at 12 14 16 PM" src="https://github.com/Expensify/App/assets/766197/5d059a86-c8fb-4c27-86f2-74d2ccceccda">

parasharrajat reviewed (changes requested):
More conflicts @BrtqKr

trjExpensify commented on src/languages/en.ts:
```diff
@@ -2970,6 +2983,34 @@ export default {
+                exportedToIntegration: ({label}: ExportedToIntegrationParams) => `exported this report to ${label}`,
+                forwarded: ({amount, currency}: ForwardedParams) => `approved ${currency}${amount}`,
+                // verify errorMessage
+                integrationsMessage: ({errorMessage}: IntegrationsMessageParams) => `failed to export this report to NetSuite. ${errorMessage}`,
```
Just checking but `NetSuite` here doesn't seem to be variable depending on what accounting solution is connected? I.e they could be exporting to Xero, not NetSuite, for example.

BrtqKr committed:
add label to integrationsMessage

BrtqKr committed:
remove todo

BrtqKr committed:
Merge remote-tracking branch 'origin/main' into brtqkr/42923-localize-system-messages

BrtqKr committed:
add condition for no field

BrtqKr commented:
@trjExpensify I've added a template label as in `exportedToIntegration`

@parasharrajat I've added a conditional case for that 

`changeFieldEmpty: ({newValue, fieldName}: ChangeFieldParams) => `changed ${fieldName} to ${newValue}`,` 

@deetergp is it possible that `policy` or `type` could be empty in some kind of scenario?

parasharrajat reviewed (approved):
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

:ribbon: :eyes: :ribbon:  C+ reviewed

deetergp reviewed (approved):
PR Looks good to me too 👍

deetergp commented:
> @deetergp is it possible that policy or type could be empty in some kind of scenario?

No, I don't think that is possible.

deetergp reviewed (changes requested):
@BrtqKr The perf-tests are failing consistently due to `FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory` and I believe it is related to the changes in tests/utils/collections/reportActions.ts. We ran into a similar issue in an [earlier PR](https://github.com/Expensify/App/pull/42811/files) and those changes are what resolved it. Can you undo those changes?

BrtqKr committed:
Merge remote-tracking branch 'origin/main' into brtqkr/42923-localize-system-messages

BrtqKr committed:
revert tests

BrtqKr committed:
revert tests

BrtqKr commented:
@deetergp I've merged main and reverted tests it seems to be passing the check

deetergp reviewed (approved):
Looks good! Thanks for the changes 👍

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/9783702773) to staging by https://github.com/deetergp in version: 9.0.4-0 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Cherry-picked](https://github.com/Expensify/App/actions/runs/9800117961) to staging by https://github.com/tgolen in version: 9.0.4-5 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

@Expensify/applauseleads please QA this PR and check it off on the [deploy checklist](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3AStagingDeployCash) if it passes.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/9872741083) to production by https://github.com/Julesssss in version: 9.0.5-13 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/9945901401) to production by https://github.com/thienlnam in version: 9.0.6-8 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅
